### ISSUE TITLE
Cannot run suite with <before> having <requiredEntity>

### ISSUE THREAD

**Issue by gujjwal00** [magento/magento2-functional-testing-framework#302]
I am trying to group my tests in a suite to share entities created  in `<before/>` but these entities are not being created properly.
Problem only happen if `<createData/> `has a  `<requiredEntity/>` inside it. 

Here is an example:  (I tested it with default WYSIWYGDisabledSuite )

```xml
<suite name="WYSIWYGDisabledSuite">
        <before>
            <createData entity="SimpleSubCategory" stepKey="createCategory"/>
            <createData entity="SimpleProduct" stepKey="createSimpleProduct">
                <requiredEntity createDataKey="createCategory"/>
            </createData>
        </before>

        <after>
            <deleteData createDataKey="createSimpleProduct" stepKey="deleteProduct"/>
            <deleteData createDataKey="createCategory" stepKey="deleteCategory"/>
        </after>
        <include>
            <group name="WYSIWYGDisabled"/>
        </include>
        <exclude>
            <group name="skip"/>
        </exclude>
    </suite>

```

It generates following  precondition code in Suite file:
```php
       PersistedObjectHandler::getInstance()->createEntity(
                    "createSimpleProduct",
                    "suite",
                    "SimpleProduct"
                );
```

When I run this suit all of tests tests fail with following error:

> INCOMPLETE: Error HTTP response code: 400    Response:{"message":"Error occurred during \"custom_attributes\" processing. The \"category_ids\" attribute code doesn't have a value set. Enter the value and try again."}

When I move my `<before/>` to test files, following precondition in generated for Simple product in test file:
```php
     PersistedObjectHandler::getInstance()->createEntity(
			"createSimpleProduct",
			"hook",
			"SimpleProduct",
			["createCategory"],
			null
		);
```
and all tests execute successfully.

### Preconditions

1. Magento version 2.3.0
2. Ubuntu 18.04

### Expected result

1. Test should run successfully.

### Actual result

1. All Tests fail

**Comment by jilu1** [comment]
Hello @gujjwal00, Thank you for your reporting! With latest develop branch, I saw a fatal error using your suite example. I've created an internal ticket https://jira.corp.magento.com/browse/MQE-1861 for this bug. We will keep you updated with progress and feedback through this PR.

**Comment by jilu1** [comment]
Released in 3.1.0
